### ISSUE TITLE
Port gtk2 module to gtk3

### ISSUE THREAD

**Issue by eszlari** [mltframework/mlt#472]
The only software that I know of that uses the gtk2 module (to import images) is flowblade (https://github.com/jliljebl/flowblade). Also @j-b-m told me to keep the gtk2 module for the flatpak of kdenlive to support old project files.

I know this has no high priority since Shotcut and Kdenlive use Qt for reading images, but would make packaging easier.

**Comment by jliljebl** [comment]
With Flowblade the Python 2 -> Python 3 port is coming next and we need to move on from SDL1.2 too.  I can say that it will take at least about year before I can look into contributing to this.

I do agree that we will need to do this eventually, but with high profile Gtk2 applications remaining in distributions, I do not think that we are in hurry with this.

**Comment by eszlari** [comment]
> I do agree that we will need to do this eventually, but with high profile Gtk2 applications remaining in distributions, I do not think that we are in hurry with this.

Inkscape 1.0 and Gimp 3.0 are not far off. The only remaining high profile app that still uses gtk2 that comes to my mind is Ardour.

**Comment by sandsmark** [comment]
It seems like the gtk part of the gtk module just is an extremely simple wrapper around `sdl_preview` (basically just setting the `SDL_WINDOWID` environment variable), might be better if flowblade did that themselves? Another issue is that the gtk code seems like it only works on X11 and Windows (so not Wayland for example).

The rest of the code in the gdk module doesn't seem to depend on gtk2, just gdk and pango (not very familiar with gtk, but I assume those are fairly independent and work with gtk3).

**Comment by jliljebl** [comment]
Flowblade onlý uses "pixbuf" for image producers, we don't use "gtk2_preview", "gtkrescale" or "pango".

If  "pixbuf" compiles with gtk3 and creates a functional image producer from my P.O.V) we can drop all the others. I have never seen "gtk2_preview" or "gtkrescale" been used by anyone, "pango" might be used by someone, but if that can be ported to Gtk3 then we could keep  "pixbuf" and "pango", and maybe just drop  "gtk2_preview", "gtkrescale".

I have just been working creating an SDL2 producer which succeeded but it has horrible performance even with "opengl" driver and accelerated 2D rendering reported. Forcing software renderer gives almost the same performance. So I will be continue to work towards a replacement for SDL1.2 video display but once I have that I can do the Gtk3 port.

**Comment by sandsmark** [comment]
gtkrescale is actually just gdk, the only actual gtk2 code is in gtk2_preview. It's also fairly easy to split into separate modules.

I just tested, and splitting into separate modules and removing gtk2 from the pango and gdk code seems to at least build fine.

**Comment by sandsmark** [comment]
If it is interesting, I can open a PR: https://github.com/sandsmark/mlt/commit/c55d777c92ffe569a22849685a80629fc459f113

**Comment by jliljebl** [comment]
Ok, I'll build and test this this week some day and comment here.

**Comment by ddennedy** [comment]
I agree to remove the gtk2 consumer. However, I would like to keep others in a single module with the name `gtk` since they are based on the GTK+ family of technologies.

**Comment by jliljebl** [comment]
Ok, tested the "gdk" module "pixbuf" producer and it works. I propose that we delete "gtk2" module and just have "gdk" module going forward. @sandsmark can you do the PR?

**Comment by sandsmark** [comment]
https://github.com/mltframework/mlt/pull/544

**Comment by sandsmark** [comment]
I named it `gdk` to make it obvious that it doesn't depend on GTK (both because GDK itself is more cross-platform, and because some people seem to have a generic GTK aversion).

**Comment by ddennedy** [comment]
Since gtk2 module is being deprecated, it will be removed and resolving this.
